Date  Tue, 26 Apr 2005 16:19:01 +0200
From  Marcello Maggioni <>
Subject  [PATCH] Timeout at bootTime with NEC3500A (and others) when inserted a CD in it.

Hi all,

I've attached a patch intended for solving boottime issues with this
and other drives when a CD/DVD is inserted .

Problem: Some drives (NEC 3500 , TDK 1616N , Mad-dog MD-16XDVD9, RICOH
MP5163DA , Memorex DVD9 drive and IO-DATA's too for sure) , if a
CD/DVD is inserted into the tray when the system is booted and if
before the OS bootup the BIOS checked for the presence of a bootable
CD/DVD into the drive , during the IDE probe phase the drive may
result busy and remain so for the next 25/30 seconds . This cause the
drive to be skipped during the booting phase and not begin usable
until the next reboot (if the reboot goes well and the drive doesn't
timeout again ).

Solution: Rising the timeout time from 10 seconds to 35 seconds
(during these 35 seconds  every drive should wake up for sure
according to the tests I've done) .

--- a/drivers/ide/ide-probe.c	2005-04-26 15:04:46.000000000 +0200
+++ b/drivers/ide/ide-probe.c	2005-04-26 15:04:14.000000000 +0200
@@ -638,13 +638,13 @@
 	SELECT_DRIVE(&hwif->drives[0]);
 	hwif->OUTB(8, hwif->io_ports[IDE_CONTROL_OFFSET]);
 	mdelay(2);
-	rc = ide_wait_not_busy(hwif, 10000);
+	rc = ide_wait_not_busy(hwif, 35000);
 	if (rc)
 		return rc;
 	SELECT_DRIVE(&hwif->drives[1]);
 	hwif->OUTB(8, hwif->io_ports[IDE_CONTROL_OFFSET]);
 	mdelay(2);
-	rc = ide_wait_not_busy(hwif, 10000);
+	rc = ide_wait_not_busy(hwif, 35000);
 
 	/* Exit function with master reselected (let's be sane) */
 	SELECT_DRIVE(&hwif->drives[0]);

--- a/drivers/ide/ide-probe.c	2005-04-26 15:04:46.000000000 +0200
+++ b/drivers/ide/ide-probe.c	2005-04-26 15:04:14.000000000 +0200
@@ -638,13 +638,13 @@
 	SELECT_DRIVE(&hwif->drives[0]);
 	hwif->OUTB(8, hwif->io_ports[IDE_CONTROL_OFFSET]);
 	mdelay(2);
-	rc = ide_wait_not_busy(hwif, 10000);
+	rc = ide_wait_not_busy(hwif, 35000);
 	if (rc)
 		return rc;
 	SELECT_DRIVE(&hwif->drives[1]);
 	hwif->OUTB(8, hwif->io_ports[IDE_CONTROL_OFFSET]);
 	mdelay(2);
-	rc = ide_wait_not_busy(hwif, 10000);
+	rc = ide_wait_not_busy(hwif, 35000);
 
 	/* Exit function with master reselected (let's be sane) */
 	SELECT_DRIVE(&hwif->drives[0]);
